backend/fms_core/management/commands/_delete_individual.py

```diff
@@ -0,0 +1,86 @@
+#   USER_ID: 5
+# }
+
+# This curation deletes only individual that are not referenced. Remove individual from samples references using "update_field_value" first.
+# To delete parents, a first curation need to be run to delete the child individual (or an update), then parents will be unreferenced.
+
+# function that checks the references to an individual and list them.
+def references(individual):
+    links = []
+    children = list(individual.mother_of.all())
+    if children:
+        links.append(f"Mother of {[child.name for child in children]}.")
+        children = None
+    children = list(individual.father_of.all())
+    if children:
+        links.append(f"Father of {[child.name for child in children]}.")
+        children = None
+    samples = list(individual.samples.all())
+    if samples:
+        links.append(f"Has samples {[sample.id for sample in samples]}")
+    return links
+
+
+def delete_individual(params, objects_to_delete, log):
+    log.info("Action [" + str(params[CURATION_INDEX]) + "] Update Field Value started.")
+    log.info("Comment [" + str(params.get(COMMENT, "None")) + "].")
+    log.info("Individual names : " + str(params[INDIVIDUAL_NAMES]) + ".")
+    log.info("Requester id : " + str(params.get(USER_ID)))
+
+    # initialize the curation
+    curation_code = params.get(CURATION_INDEX, "Invalid index")
+    error_found = False
+    name_array = params[INDIVIDUAL_NAMES]
+    user_id = params.get(USER_ID)
+
+    try:
+        individual_model = apps.get_model("fms_core", "Individual")
+        count_deletes = 0
+        for name in name_array:
+            try:
+                individual = individual_model.objects.get(name=name)
+                links = references(individual)
+                if links:
+                    log.error(f"Individual [{name}] is still referenced. {links}")
+                    error_found = True
+                else:
+                    log.info(f"Deleted [Individual] name [{individual.name}] id [{individual.id}]].")
+                    individual.deleted = True
+                    if user_id:
+                        individual.save(requester_id=user_id) # save using the id of the requester
+                    else:
+                        individual.save()# save using the default admin user
+                    objects_to_delete.append(individual)  # Delay deletion until after the revision block so the object get a version
+                    count_deletes += 1
+            except individual_model.DoesNotExist:
+                log.error(f"No individual found for name [{name}].")
+                error_found = True
+            except individual_model.MultipleObjectsReturned:
+                log.error(f"Multiple individuals found for name [{name}]. Provide a unique identifier.")
+                error_found = True
+    except LookupError:
+        log.error("Model [Individual] does not exist.")
+        error_found = True
+    if not error_found:
+        curation_code = None
```

Comment:
This also refers to my comment about error handling. It's better not using boolean and curation_index as a way to check if it generates errors..

Reply:
This is a way to support both exceptions and "normal" errors ... I will not throw exceptions or define custom errors for all the case where some condition raise an error. This also let us manage error much more nimbly than the exception handling approach you suggest that basically crashes your application at the first error encountered. The way around this will require in any case that kind of error handling ...

Reply:
You can raise an exception and catch it by adding a specific message to the console.. you do not have to interrupt the flow of the whole process

Reply:
if you do that your transaction will complete and you'll have a partial curation ...